tests/main/snap-service-stop-mode/task.yaml

```diff
@@ -19,6 +19,8 @@ execute: |
 
     echo "When the service snap is installed"
     install_local test-snapd-service
+    # give services a bit of time to setup signal handlers
+    sleep 1
```

Comment:
Can we make this more deterministic?  Looks like something that might break every once in a while.

Reply:
This is because systemd thinks the service is up before it is really operational. This is an inherent flaw of the `daemon: simple` mode that is only fixed with `daemon: notify`. Unfortunately we cannot use `daemon: notify` until we allow writes to `$NOTIFY_SOCKET` (which is used by the sd_notify(3) mechanism).

I have prepared a patch that unblocks `sd_notify` and uses it instead of the sleep calls. I will post updates as I get results.

Reply:
This is a bit of a chicken-and-egg that I'm still exploring since `daemon-notify` is not auto-connected. I'm making experiments to see if the test is buggy now if everything is switched over to sd-notify and the plug auto-connected (just for the experiment).

Reply:
My experiment yielded some interesting results that I will summarize here:
- the test is racy because systemd considers the service to be active before the service process can establish signal handlers
- using `daemon: notify` and the `daemon-notify` interface is difficult today
- the test works flawlessly without the `sleep 1` calls if we use an external mechanism to wait for the service to start (or use the systemd notification on a snapd patched to auto-connect that interface)

Reply:
I now applied a workaround based on writing a file that indicates service readiness.